### PR TITLE
Move GLSL compilation to build time

### PR DESCRIPTION
**Pull request by kainino0x**
This compiles GLSL using a build time macro(!) via babel-plugin-macros, instead of doing shader compilation online. Rationale:

1. It's too annoying to check my current non-size-optimized 900k binary into Chrome and WPT.
2. Even though we'll most likely need to be able to generate shaders programmatically later on, we can get away without runtime compilation for now.
3. We probably aren't going to use it in the long run, so hopefully by the time we need to replace it we'll know what the shading language is.

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -1,20 +1,23 @@
...
+  "plugins": [
+    "@babel/plugin-proposal-class-properties",
+    "@babel/plugin-syntax-dynamic-import",
+    "macros",
+    [
+      "module-resolver",
+      {
+        "root": ["./src"],
+        "alias": {
+          "@webgpu/shaderc": "./src/shaderc.js"
```
Comment (austinEng) [inline .babelrc]: I don't think we need shaderc anymore

File: src/tools/glsl.macro.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
+const { createMacro, MacroError } = require('babel-plugin-macros');
```
Comment (austinEng) [inline src/tools/glsl.macro.js]: nit: Can this be `import`s?

File: src/tools/glsl.macro.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+const glslangModule = require('@webgpu/glslang');
+
+let glslang;
+function initializeGlslang() {
+  if (!glslang) {
+    glslang = glslangModule();
+  }
+}
+
+module.exports = createMacro(({ references, state, babel }) => {
```
Comment (austinEng) [inline src/tools/glsl.macro.js]: nit: Can this work with `export default`?

**Comment by austinEng** [comment]
Another thought as I was reviewing the Chromium CL: Can we have the macro insert a comment with the source text of the compiled shader code?

File: src/tools/glsl.macro.js (hunk 2 above)
Comment (kainino0x) [reply src/tools/glsl.macro.js]: Unfortunately it's not easy to get this code to run as an ES6 module. It's running in node as a Babel plugin and I can't figure out how to make it allow that.

Comment (austinEng) [reply src/tools/glsl.macro.js]: Ah yes, makes sense.

**Pull request merged**